examples/generative/molecule_generation.py

```diff
@@ -0,0 +1,493 @@
+        super(Decoder, self).__init__(**kwargs)
+        self.linear_1 = layers.Dense(latent_dim, activation="selu")
+        self.repeat_vector = layers.RepeatVector(max_length)
+        self.gru_1 = layers.GRU(701, dropout=0.0, return_sequences=True)
+        self.gru_2 = layers.GRU(701, dropout=0.4, return_sequences=True)
+        self.gru_3 = layers.GRU(901, dropout=0.0, return_sequences=True)
+        self.linear_2 = layers.TimeDistributed(layers.Dense(charset_length))
+
+    def call(self, x):
+        x = self.linear_1(x)
+        x = self.repeat_vector(x)
+        x = self.gru_1(x)
+        x = self.gru_2(x)
+        x = self.gru_3(x)
+        x = self.linear_2(x)
+        x = tf.keras.activations.softmax(x, axis=2)
+
+        return x
+
+
+"""
+## Build Sampling layer
+"""
+
+
+class Sampling(layers.Layer):
+    """Uses (z_mean, z_log_var) to sample z, the vector encoding."""
+
+    def call(self, inputs):
+        z_mean, z_log_var = inputs
+        batch = tf.shape(z_log_var)[0]
+        dim = tf.shape(z_log_var)[1]
+        epsilon = tf.keras.backend.random_normal(shape=(batch, dim))
+        return z_mean + tf.exp(0.5 * z_log_var) * epsilon
+
+
+"""
+## Building the VAE
+With this model we try to optimize three losses:
+
+
+* Categorical Crossentropy Loss
+* KL Loss
+* Property prediction Loss
+
+The Categorical Crossentropy loss is to give a measure of the model's reconstruction
+capacity. The Property prediction loss is to estimate the mean squared error of the
+predicted and actual property after passing the ```z_mean(generated from the encoder)```
+through a property prediction model or property prediction layer.
+"""
+
+
+class MoleculeGenerator(tf.keras.Model):
+    def __init__(self, encoder, decoder, max_len, **kwargs):
+        super(MoleculeGenerator, self).__init__(**kwargs)
+        self.encoder = encoder
+        self.decoder = decoder
+        self.property_prediction_layer = layers.Dense(1)
+        self.max_len = max_len
+
+        self.train_total_loss_tracker = tf.keras.metrics.Mean(name="train_total_loss")
+        self.train_reconstruction_loss_tracker = tf.keras.metrics.Mean(
+            name="train_reconstruction_loss"
+        )
+        self.train_kl_loss_tracker = tf.keras.metrics.Mean(name="train_kl_loss")
+        self.train_pp_loss_tracker = tf.keras.metrics.Mean(
+            name="train_property_prediction_loss"
+        )
+
+        self.val_total_loss_tracker = tf.keras.metrics.Mean(name="val_total_loss")
+        self.val_reconstruction_loss_tracker = tf.keras.metrics.Mean(
+            name="val_reconstruction_loss"
+        )
+        self.val_kl_loss_tracker = tf.keras.metrics.Mean(name="val_kl_loss")
+        self.val_pp_loss_tracker = tf.keras.metrics.Mean(
+            name="val_property_prediction_loss"
+        )
+
+    def compile(self, vae_optimizer, **kwargs):
+        super().compile(**kwargs)
+        self.vae_optimizer = vae_optimizer
```

Comment:
Why not just use `optimizer`? Why override `compile()`?

Reply:
i have removed this snippet, I also felt its unnecessary